### ISSUE TITLE
Opening a ContextMenu programmatically doesn't trigger the ContextMenuOpening event

### ISSUE THREAD

**Issue by andrei828** [dotnet/wpf#8879]
### Description

When a `ContextMenu` is opened programmatically by setting the `IsOpen` property to `true`, the `OnContextMenuOpening` handler is never called.

### Reproduction Steps

XAML:
```xaml
<Window x:Class="WpfApp1.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d"
        Title="MainWindow" Height="400" Width="400">

    <Window.ContextMenu>
        <ContextMenu Name="ctxMenu">
            <MenuItem Header="Testing"/>
        </ContextMenu>
    </Window.ContextMenu>
    <Button Click="Button_Click" Width="20" Height="20"/>
</Window>
```

Code-behind
```cs
using System.Windows;
using System.Windows.Controls;

namespace WpfApp1
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
        }

        protected override void OnContextMenuOpening(ContextMenuEventArgs e)
        {
            base.OnContextMenuOpening(e);
        }

        private void Button_Click(object sender, RoutedEventArgs e)
        {
            ctxMenu.IsOpen = true;
        }
    }
}
```

Clicking the button opens up the `ContextMenu` of the Window programmatically. Notice that the `OnContextMenuOpening` handler is never called.

### Expected behavior

The ContextMenu opens and the `OnContextMenuOpening` is called the same way it works when right-clicking to open it. 

### Actual behavior

The ContextMenu opens but the `OnContextMenuOpening` method is never called.

### Regression?

I also tested on a project with the `.NET Framework 4.8` Target Framework and the issue is still reproducible.

### Configuration

TargetFramework: net8.0-windows
OS: Windows 11 Enterprise, x64
Also reproducible on  .NET Framework 4.8

**Comment by miloush** [comment]
Yeah so the `ContextMenuOpening` and `ContextMenuClosing` belong to `ContextMenuService`. They are triggered when it decides to open or close the menu, as a result of key press or mouse button press. Once you get a chance to handle them, it proceeds to setting the `IsOpen` property.

You setting the `IsOpen` property yourself is basically after these events would have normally occurred. If you were handling `ContextMenuOpening` and wanted a menu open as a result, you would mark it as handled and set `IsOpen` inside your handler, so it doesn't make much sense it would trigger the event again.

You could treat these events as "something else decided to open/close my menu, are you happy with that?". I agree the documentation is slightly misleading.

**Comment by czdietrich** [comment]
`ContextMenuOpening` and `ContextMenuClosing` can be used to implement a fully customized _context menu_ without using the built-in `ContextMenu`, so you have events to hook up your custom _context menu_ behavior. 
Still, the events should maybe better be named something like `RequestContextMenuOpen` and `RequestContextMenuClose`.
